### PR TITLE
Documentation,manifests: Document tuning metering components

### PR DESCRIPTION
**Pull request by chancez**
This is the first pass at documenting how to properly tune metering.
It still lacks of lot of concrete answers to things like how to calculate storage or max memory for a component based on cluster size, but it's better than nothing, and provides guidance on how to understand what values to use.

Link to main doc: https://github.com/chancez/operator-metering/blob/document_resource_tuning/Documentation/tuning.md

### CODE REVIEW

File: Documentation/tuning.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+The reporting-operator's most resource intensive task is querying Prometheus for metrics, storing the results in memory, converting them into a SQL insert statement, and batching those inserts to Presto.
+Performing imports is mostly a memory and IO intensive task, and thus increasing memory beyond defaults can help when dealing with a lot of metrics on larger or highly active clusters.
+
+### Presto
+
+Presto is the main component that does real computation intensive work.
+Additionally, it processes everything in memory when a query comes in, meaning it's primary resource constraint will often be memory, and CPU can be a problem when there's a lot of garbage collections occurring while processing a query.
+
+Presto itself doesn't store any data, and instead access data on demand when a query is executed.
+This means network is often a bottleneck, and can be something you may need to consider.
```
Comment (EmilyM1) [inline Documentation/tuning.md]: Presto itself doesn't store the data, instead it accesses the data on demand ...

File: Documentation/tuning.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Resource Tuning
+
+Often times the default resource requests/limits set by Metering aren't sufficient for larger clusters with more resources changing and producing more metrics.
+Without proper resource limits, pods are unable to correctly operate without experiencing pod restarts or crashes (often do to the pod being OOMKilled).
+
```
[the remark this answers is not in the document]
Comment (chancez) [reply Documentation/tuning.md]: Good catch

File: Documentation/tuning.md (hunk 1 above)
Comment (chancez) [reply Documentation/tuning.md]: Good catch

File: Documentation/tuning.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Hive metastore is often fairly idle, and is queried everytime Presto must interact with a table.
+These interactions are all fairly light weight, and thus the amount of resources the metastore requires isn't very large.
+However, it does retain quite a bit of information in memory, and can require more memory than the default values provide when it's been running for a longer period of time.
+
+The Hive server component is very lightly used and is only interacted with when creating new tables, or modifying partitions of an AWS Billing ReportDataSource table, and thus requires fewer resources.
+
+### HDFS
+
+By default, Metering installs HDFS for storage. While the amount of data isn't large in most cases, you do want to consider running multiple HDFS datanode replicas for redundancy.
+Because HDFS is commonly accessed by Presto when data is being stored and queried, it's also typically a component which more resources over time.
```
Comment (pruan-rht) [inline Documentation/tuning.md]: I think this sentence is missing a verb...should it be `typically a component which uses more resources over time`?

**Comment by chancez** [comment]
@pruan-rht Fixed those items. Good catches!

File: Documentation/tuning.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# Resource Tuning
+
+Often times the default resource requests/limits set by Metering aren't sufficient for larger clusters with more resources changing and producing more metrics.
+Without proper resource limits, pods are unable to correctly operate without experiencing pod restarts or crashes (often due to the pod being OOMKilled).
+
+Here's a common list of reasons you may need to increase your resource limits:
+
+- Larger clusters. If you're running more than 10 nodes, it's likely the defaults will insufficient. The defaults are set low to make it easier to install onto smaller clusters.
+- Pod's being OOMKilled. Check if any pod has a restart count that is increasing over time. Use `kubectl describe pod` to determine why it restarted. If it's due to OOMKilled, then this pod needs more memory.
```
Comment (tschuy) [inline Documentation/tuning.md]: remove `'`

File: manifests/metering-config/recommended-resource-limits.yaml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        # minutes of data is more than a few hundred metrics per time series.
+        # This comes up when you have hundreds of namespaces and pods.
+        promsumChunkSize: "5m"
+        promsumPollInterval: "5m"
+        # This is the resolution of the metrics. Setting this higher reduces
+        # granularity of the data we import, but also reduces the number of
+        # metrics returned per query.
+        promsumStepSize: "60s"
+
+      # Increasing memory to 500Mi allows reporting-operator to keep more
```
Comment (tschuy) [inline manifests/metering-config/recommended-resource-limits.yaml]: this should probably be the same as the number listed in the actual config (standardize to either 500 or 512; elsewhere you mainly use multiples of 5 so I'd recommend 500)

Comment (chancez) [reply manifests/metering-config/recommended-resource-limits.yaml]: Whoops. I was messing with the different unit suffixes. Good catch.

**Pull request merged**